samza-api/src/main/java/org/apache/samza/table/descriptors/RemoteTableDescriptor.java

```diff
@@ -112,6 +114,7 @@ public RemoteTableDescriptor(String tableId) {
   public RemoteTableDescriptor<K, V> withReadFunction(TableReadFunction<K, V> readFn) {
     Preconditions.checkNotNull(readFn, "null read function");
     this.readFn = readFn;
+    this.readFn.setTableId(String.format("%s.%s", tableId, READ_FN));
```

Comment:
This allows the `RemoteTableDescriptor` to set the `tableId` for `TableReadFunction` and `TableWriteFunction` and other `TablePart`s. So that when `TableReadFunction` implements `toConfig` method, it knows it's `tabldId`.
This makes it easier for users since they don't need to type in same `tableId` several times when using `RemoteTableDescritptor`.